components/AddComment.jsx

```diff
@@ -0,0 +1,51 @@
+import { useState } from "react";
+import { postCommentByArticleId } from "../api/api";
+
+export default function AddComment(props) {
+  const [newComment, setNewComment] = useState("");
+  const [validation, setValidation] = useState(false);
+  const [failedComment, setFailedComment] = useState(false);
+  const username = "jessjelly";
+
+  function createNewComment(e) {
+    e.preventDefault();
+    if (validation) {
+      postCommentByArticleId(props.article_id, { username: username, body: newComment })
+        .then(() => {
+          setFailedComment(false);
+        })
+        .catch(() => {
+          setFailedComment(true);
+          props.setComments((currComments) => [...currComments.slice(1)]);
+        });
+      props.setComments((currComments) => {
+        const createComment = {
```

Comment:
Check what your back end is expecting and what the database will do automatically. What did you design your tests to accept?